### PR TITLE
Burial date warning

### PR DESCRIPTION
**Pull request by cvalarida**
Connects to https://github.com/department-of-veterans-affairs/vets.gov-team/issues/3832

image

image

image

### CODE REVIEW

File: src/js/burials/config/form.js [hunk 1]
```diff
@@ -182,6 +183,20 @@ const formConfig = {
           uiSchema: {
             deathDate: currentOrPastDateUI('Date of death'),
             burialDate: currentOrPastDateUI('Date of burial (includes cremation or interment)'),
+            'view:burialDateWarning': {
+              'ui:description': burialDateWarning,
+              'ui:options': {
+                hideIf: formData => {
+                  // If they haven't entered a complete year, don't jump the gun and show the warning
+                  if (formData.burialDate && formData.burialDate.split('-')[0].length !== 4) {
```
Comment (jbalboni) [inline src/js/burials/config/form.js]: I think I did this by matching a regex somewhere else. What do you think of doing that instead?

**Review by jbalboni — APPROVED** [review]
Looks good to me. Had one question/comment.

File: src/js/burials/config/form.js (hunk 1 above)
Comment (cvalarida) [inline src/js/burials/config/form.js]: Lemme whip up a benchmark and get back to you!

Comment (cvalarida) [inline src/js/burials/config/form.js]: image
Looks like regex is the way to go. Good call.

**Review by cehsu — APPROVED** [review]

File: src/js/burials/config/form.js [hunk 2]
```diff
@@ -182,6 +183,20 @@ const formConfig = {
...
             burialDate: currentOrPastDateUI('Date of burial (includes cremation or interment)'),
+            'view:burialDateWarning': {
+              'ui:description': burialDateWarning,
+              'ui:options': {
+                hideIf: formData => {
+                  // If they haven't entered a complete year, don't jump the gun and show the warning
+                  if (formData.burialDate && !/\d{4}-\d{1,2}-\d{1,2}/.test(formData.burialDate)) {
+                    return true;
+                  }
+
```
Comment (cehsu) [inline src/js/burials/config/form.js]: Is the linter fine with this spacing?

Comment (cvalarida) [inline src/js/burials/config/form.js]: Yeah, it only complains at more than 2 blank lines.

**Pull request merged**